### PR TITLE
items/routeros: purge unused subitems

### PR DESCRIPTION
**Pull request by MaZderMind**
this works the same way that directories work.

### CODE REVIEW

File: bundlewrap/items/routeros.py [hunk 1]
```diff
@@ -113,25 +163,32 @@ def run_routeros(self, *command):
         })
         return result
 
-    def _add(self, command, kwargs):
-        identifier = self.name.split("?", 1)[1]
+    def parase_identifier(self, identifier):
```
Comment (trehn) [inline bundlewrap/items/routeros.py:166]: typo

File: bundlewrap/items/routeros.py [hunk 2]
```diff
@@ -26,30 +29,68 @@ def __repr__(self):
...
         del cdict['delete']
+        del cdict['purge']
         return cdict
 
+    def get_basename(self, name):
+        return name.split("?", 1)[0]
+
+    @property
+    def basename(self):
+        return self.get_basename(self.name)
```
Comment (fkusei) [inline bundlewrap/items/routeros.py:53]: Is there any reason not to merge those two methods? `get_basename()` is not used anywhere else? (same for `get_identifier()` below.

Comment (MaZderMind) [reply bundlewrap/items/routeros.py:53]: `get_identifier` *is* used further down in `_get_subitems_to_purge` and I split up ´basename` for symmetry, too.

File: bundlewrap/items/routeros.py [hunk 3]
```diff
@@ -26,30 +29,68 @@ def __repr__(self):
...
+    def identifier(self):
+        return self.get_identifier(self.name)
+
     def fix(self, status):
+        if status.sdict:
+            for subitem_id, subitem_name in status.sdict.get('subitems_to_purge', set()):
+                self._remove(self.basename, subitem_id)
+
         if status.must_be_created:
-            self._add(self.name.split("?", 1)[0], status.cdict)
```
Comment (MaZderMind) [inline bundlewrap/items/routeros.py:38]: cdict/sdict mixed up

File: docs/content/items/routeros.md [hunk 4]
```diff
@@ -16,13 +16,23 @@ Manages RouterOS configuration through the API (port 8728). You can address ever
         "/interface/vlan?name=vlan7": {
             "delete": True,
         },
+        "/interface/vlan": {
+            "purge": {
+                "id-by": "name",
```
Comment (vain) [inline docs/content/items/routeros.md:21]: The `purge` mechanism could use some more documentation. This is something that we came up with, so we can’t refer people to Mikrotik docs.

The docs should answer the following questions:

- How is `id-by` used? Which value must I use, how do I find it?
- What’s the meaning of `keep/dynamic`? Are there more options for `keep` other than `dynamic`?

Comment (MaZderMind) [reply docs/content/items/routeros.md:21]: Indeed, I will extend the docs on this.

Comment (MaZderMind) [reply docs/content/items/routeros.md:25]: Would you mind re-reading the documentation? Thanks in advance!

**Comment by MaZderMind** [comment]
🔔 *push* Would love to get this merged soon.

**Comment by trehn** [comment]
Thank you!

**Comment by MaZderMind** [comment]
Wheee \o/

**Pull request merged**